static/js/http.js

```diff
@@ -12,14 +12,16 @@ function request (method, path, options = {}) {
     headers.append('Content-Type', 'application/json')
     opts.body = JSON.stringify(body)
   }
-  return new Promise((resolv, reject) => {
-    window.fetch(path, opts)
-      .then(response => {
-        if (!response.ok)
-          return response.json().then(json => reject(new HTTPError(response.status, json.reason)))
-        return resolv(response.json())
-      })
-  })
+  return window.fetch(path, opts)
+    .then(response => {
+      if (!response.ok) {
+        return response.json().then(json => { throw new HTTPError(response.status, json.reason) })
+      }
+      if (!response.headers.has('content-length') || parseInt(response.headers.get('content-length')) === 0) {
```

Comment:
All json responses has transfer-encoding: chunked, and hence no content-length.

Reply:
Lol, bad testing. But yes, i see now. It was the 201 that had a content-length=0. 

Maybe it's safe to assume we never have a body for 201 and 204 then? It was my first idea, but 201 may contain a body (according to specification, if I'm not mistaken).